### PR TITLE
Implement VerifyAndRegisterAlert and MhvBasicAccountAlert to MHV Landing page - version 2

### PR DESCRIPTION
**Pull request by jzucadi**
## Summary

- Adds `VerifyAndRegisterAlert` to MHV Home
- Adds `MhvBasicAccountAlert` to MHV Home
- Creates a new alert component called `LandingPageAlerts` to encapsulate all of the alerts that appear on the MHV landing page

## Related issue(s)

1. department-of-veterans-affairs/va.gov-team#77357
2. department-of-veterans-affairs/va.gov-team#83069

## MHV Basic Account

|         | Before | After |
| ------- |------ | ----- |
| Mobile  |   Screenshot 2024-07-17 at 2 22 46 AM  | Screenshot 2024-08-01 at 1 52 41 PM Screenshot 2024-08-01 at 1 52 49 PM |
| Desktop  |  Screenshot 2024-07-17 at 2 22 13 AM      |  Screenshot 2024-08-01 at 1 51 50 PM |

## Verify and Register Alert

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |  Screenshot 2024-07-17 at 2 22 46 AM      |  Screenshot 2024-07-25 at 6 23 04 PM |
| Desktop |   Screenshot 2024-07-17 at 2 22 13 AM | Screenshot 2024-07-25 at 6 23 18 PM |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

**Review by radavis — COMMENTED** [review]
I like that our alerts are now wrapped up by a single component. 

We're getting into territory where we are showing multiple alerts to the user (e.g. - loa1 & mhv case yields UnregisteredUser and MhvBasicAccountAlerts). I think we need to refactor the logic for showing these alerts before we can merge. [Here's some guidance](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/digital-health-modernization/engineering/mhv-home-alerts.md#proposed-alert-flow) for showing the most relevant alert to the user.

Screenshot 2024-07-30 at 10-17-47 My HealtheVet Veterans Affairs

**Review by micahchiang — DISMISSED** [review]

**Comment by jzucadi** [comment]
> I like that our alerts are now wrapped up by a single component.
> 
> We're getting into territory where we are showing multiple alerts to the user (e.g. - loa1 & mhv case yields UnregisteredUser and MhvBasicAccountAlerts). I think we need to refactor the logic for showing these alerts before we can merge. [Here's some guidance](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/digital-health-modernization/engineering/mhv-home-alerts.md#proposed-alert-flow) for showing the most relevant alert to the user.
> 
> Screenshot 2024-07-30 at 10-17-47 My HealtheVet Veterans Affairs

Agreed!

**Comment by dcloud** [comment]
@jzucadi I'm also seeing the three alerts per @radavis screenshot and agree that we probably want to show the single most relevant alert. Let's talk about this as a team and make sure UX is informed and on board

**Review by sterkenburgsara — COMMENTED** [review]
These two alerts shouldn't display at the same time. There is criteria in the ticket for the first alert in the stack that it should _only_ be for ID.me and Login.gov (have to remove MyHealtheVet sign-in credential from that logic). If a user signs in with a My HealtheVet basic (LOA1) credential they should see the second alert (yellow one) only.

File: src/applications/mhv-landing-page/components/LandingPageAlerts.jsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+const LandingPageAlerts = ({
+  signInService,
+  showsVerifyAndRegisterAlert,
+  unVerifiedHeadline,
+  userHasMhvAccount,
+  userHasMhvBasicAccount,
+  userRegistered,
+  userVerified,
+}) => {
+  if (userHasMhvBasicAccount && !userVerified) {
```
[the remark this answers is not in the document]
Comment (dcloud) [reply src/applications/mhv-landing-page/components/LandingPageAlerts.jsx:18]:
```suggestion
  if (userHasMhvBasicAccount) {
```

File: src/applications/mhv-landing-page/components/LandingPageAlerts.jsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    return <MhvBasicAccountAlert />;
+  }
+
+  if (
+    (signInService === 'idme' || signInService === 'logingov') &&
+    !userVerified &&
+    showsVerifyAndRegisterAlert
```
Comment (dcloud) [inline src/applications/mhv-landing-page/components/LandingPageAlerts.jsx:25]: Doesn't `showsVerifyAndRegisterAlert` also check the sign in service is idme or logingov and is LOA3? I don't think we should need the first two conditional checks:

```suggestion
    showsVerifyAndRegisterAlert
```

**Review by dcloud — COMMENTED** [review]
I left 2 minor suggestions, but think this looks good. Seeing a merge conflict, which I think would negate a ✅ , so please ping me once that conflict has been resolved

File: src/applications/mhv-landing-page/tests/e2e/alerts/unverified.cypress.spec.js [hunk 3]
```diff
@@ -24,9 +24,16 @@ describe(appName, () => {
...
+          showVerifyAndRegisterAlert: false,
         });
         cy.injectAxeThenAxeCheck();
 
+        // Log the page content
+        cy.log('Page Content:');
+        cy.get('body').then($body => {
+          cy.log($body.text());
+        });
+
```
Comment (radavis) [inline src/applications/mhv-landing-page/tests/e2e/alerts/unverified.cypress.spec.js:36]: This should be removed.

**Review by radavis — DISMISSED** [review]
Looks good. You will need to remove the log statements from the e2e spec

File: src/applications/mhv-landing-page/tests/e2e/alerts/unverified.cypress.spec.js (hunk 3 above)
Comment (jzucadi) [reply src/applications/mhv-landing-page/tests/e2e/alerts/unverified.cypress.spec.js:36]: Sorry! I forgot to take this out.

**Review by dcloud — APPROVED** [review]
This looks good to me, thanks for leaving things nicer than you found them! 🚀

**Pull request merged**